build.gradle

```diff
@@ -213,53 +208,45 @@ publishing {
                             name 'Andreas Marek'
                         }
                     }
+                    scm {
+                        url "https://github.com/atlassian-labs/nadel"
+                    }
                 }
             }
         }
     }
 }
 
+artifactory {
+    publish {
+        contextUrl = 'https://packages.atlassian.com/'
 
-bintray {
-    user = System.env.BINTRAY_USER ?: project["bintray.user"]
-    key = System.env.BINTRAY_API_KEY ?: project["bintray.key"]
-    publications = ['graphqlJava']
-    publish = true
-    pkg {
-        repo = 'nadel'
-        name = 'nadel'
-        desc = 'GraphQL stitching DSL'
-        licenses = ['MIT']
-        vcsUrl = 'https://github.com/graphql-java/nadel'
-
-        version {
-            name = project.version
-            desc = project.description
-            released = new Date()
-            vcsTag = 'v' + project.version
-            gpg {
-                sign = true
-            }
-            mavenCentralSync {
-                sync = true
-                user = System.env.MAVEN_CENTRAL_USER
-                password = System.env.MAVEN_CENTRAL_PASSWORD
-                close = '1'
-            }
+        repository {
+            repoKey = 'maven-central'
+            username = System.env.ARTIFACTORY_USERNAME
+            password = System.env.ARTIFACTORY_API_KEY
+        }
+        defaults {
+            publications('maven')
+            publishIvy = false
+            // This needs to be "false", otherwise, the artifactory plugin will try to publish
+            // a build info file to Artifactory and fail because we don't have the permissions to do that.
+            publishBuildInfo = false
         }
     }
 }
 
+signing {
+    useInMemoryPgpKeys("$System.env.SIGNING_KEY", "$System.env.SIGNING_PASSWORD")
+    sign publishing.publications.graphqlJava
```

Comment:
does this cover all artifacts like javadoc, sources etc? In GraphQL Java it is this: https://github.com/graphql-java/graphql-java/blob/master/build.gradle#L249

Reply:
I think it's doing the same thing. In graphql-java we do `sign publishing.publications ` and here it's `sign publishing.publications.graphqlJava`. That's the only difference I see. 

Do you mean it should be `sign publishing.publications` here? I took it from this atlassian open source gradle template https://github.com/atlassian-labs/oss-gradle-template/blob/master/build.gradle#L119. I suspect it's the same syntaxis for the same thing as `publications` only contain `graphqlJava`.

Also, `graphqlJava` lists 2 artifacts: https://github.com/atlassian-labs/nadel/pull/215/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R169 `javadocJar` and `sourcesJar` so they should be covered